spec/druid_tools_spec.rb

```diff
@@ -322,6 +341,32 @@
       FileUtils.remove_entry workspace
     end
 
+    context 'with an access druid sharing the first three path segments' do
+      before do
+        # Nil the create records for this context because we're in a known read only one
+        dr1.mkdir
+        dr2.mkdir
+        dr3.mkdir
+        dr3.prune!
+      end
+
+      it 'deletes the outermost directory' do
+        expect(File).not_to exist(dr3.path)
+      end
+
+      it 'does not delete empty ancestor directories' do
+        expect(File).to exist(dr1.pruning_base)
+        expect(File).to exist(dr1.pruning_base.parent)
+      end
+
+      it 'stops at ancestor directories that have children' do
+        # 'cd/456/ef' should still exist because of dr1
+        shared_ancestor = dr1.pruning_base.parent
+        expect(shared_ancestor.to_s).to match(%r{cd/456/ef$})
+        expect(File).to exist(shared_ancestor)
+      end
+    end
```

Comment:
This is the same test change made in https://github.com/sul-dlss/dor-services-app/pull/3790